sonar-batch/src/main/java/org/sonar/batch/bootstrap/BatchComponents.java

```diff
@@ -66,8 +67,16 @@ private BatchComponents() {
       HtmlReport.class,
       IssuesReportBuilder.class,
       SourceProvider.class,
-      RuleNameProvider.class
-      );
+      RuleNameProvider.class,
+
+      // Tasks
+      Tasks.class,
+      ListTask.DEFINITION,
+      ListTask.class,
+      ScanTask.DEFINITION,
+      ScanTask.class,
+      ViewsTask.DEFINITION,
+      ViewsTask.class);
```

Comment:
The point to separate definition from implementation is to benefit from Pico lazy loading. We can load all definitions with very few risk of crash. Then try to load only the implementation of the requested task. It means a broken task (with an injected component that is not available for example) will not prevent to run other tasks.